test/he/test_conn.c

```diff
@@ -1119,23 +1119,47 @@ void test_he_internal_renegotiate_ssl_not_online(void) {
 void test_he_internal_renegotiate_ssl_already_renegotiating(void) {
   // Set our state online and set renegotiation in progress
   conn.state = HE_STATE_ONLINE;
+  conn.renegotiation_due = true;
   conn.renegotiation_in_progress = true;
 
   he_return_code_t res = he_internal_renegotiate_ssl(&conn);
   TEST_ASSERT_EQUAL(HE_SUCCESS, res);
+  TEST_ASSERT_FALSE(conn.renegotiation_due);
```

Comment:
Do you mean to test `renegotiation_in_progress` here?

Reply:
No, we want to test "he_internal_renegotiate_ssl should reset renegotiation_due and return immediately" here. I've added a comment.